tools/c7n_azure/tests_azure/tests_resources/test_subscription.py

```diff
@@ -82,6 +84,12 @@ def test_add_policy(self, definition_patch):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+        # Create Policy Assignment API might return before it was actually created
+        # Insert a delay to keep test active during nightly runs until this is fixed:
+        # https://github.com/Azure/azure-rest-api-specs/issues/7691
+        if not self.is_playback():

```

Comment:
I think a better approach is to attempt to fetch and retry w/ a 3 second sleep than to just arbitrarily sleep.